posts/2023-08-24-Rust-1.72.0.md

```diff
@@ -0,0 +1,60 @@
+
+If you have a previous version of Rust installed via rustup, you can get 1.72.0 with:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.72.0](https://github.com/rust-lang/rust/releases/tag/1.72.0) on GitHub.
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.72.0 stable
+
+### Uplifted lints from Clippy
+
+Several lints from Clippy have been pulled into `rustc`:
+
+* [`clippy::undropped_manually_drops`](https://rust-lang.github.io/rust-clippy/rust-1.71.0/index.html#undropped_manually_drops) to [`undropped_manually_drops`](https://doc.rust-lang.org/1.72.0/rustc/lints/listing/deny-by-default.html#undropped-manually-drops) (deny)
+  - `ManuallyDrop` does not drop its inner value, so calling `std::mem::drop` on it does nothing. Instead, the lint will suggest `ManuallyDrop::into_inner` first, or you may use the unsafe `ManuallyDrop::drop` to run the destructor in-place. This lint is denied by default.
+
+* [`clippy::invalid_utf8_in_unchecked`](https://rust-lang.github.io/rust-clippy/rust-1.71.0/index.html#invalid_utf8_in_unchecked) to [`invalid_from_utf8_unchecked`](https://doc.rust-lang.org/1.72.0/rustc/lints/listing/deny-by-default.html#invalid-from-utf8-unchecked) (deny) and [`invalid_from_utf8`](https://doc.rust-lang.org/1.72.0/rustc/lints/listing/warn-by-default.html#invalid-from-utf8) (warn)
+  - The first checks for calls to `std::str::from_utf8_unchecked` and `std::str::from_utf8_unchecked_mut` with an invalid UTF-8 literal, which violates their safety pre-conditions, resulting in undefined behavior. This lint is denied by default.
+  - The second checks for calls to `std::str::from_utf8` and `std::str::from_utf8_mut` with an invalid UTF-8 literal, which will always return an error. This lint is a warning by default.
+
+* [`clippy::cmp_nan`](https://rust-lang.github.io/rust-clippy/rust-1.71.0/index.html#cmp_nan) to [`invalid_nan_comparisons`](https://doc.rust-lang.org/1.72.0/rustc/lints/listing/warn-by-default.html#invalid-nan-comparisons) (warn)
+  - This checks for comparisons with `f32::NAN` or `f64::NAN` as one of the operands. NaN does not compare meaningfully to anything – not even itself – so those comparisons are always false. This lint is a warning by default, and will suggest calling the `is_nan()` method instead.
+
+* [`clippy::cast_ref_to_mut`](https://rust-lang.github.io/rust-clippy/rust-1.71.0/index.html#cast_ref_to_mut) to [`invalid_reference_casting`](https://doc.rust-lang.org/1.72.0/rustc/lints/listing/allowed-by-default.html#invalid-reference-casting) (allow)
```

Comment:
Maybe it's worth pointing out that it's currently `allow` because there are some issues with it, and it will be raised when they are fixed. People may be surprised why a "this is UB" lint is allow-by-default.

Reply:
I was trying to hint at that in the following line, although I don't really know what exactly is lacking in the lint as-is. False positives?

Reply:
@Urgau, can you clarify the reason for "allow" in brief layman's terms?
(between rust-lang/rust#113422 and rust-lang/rust#112431)

Reply:
Does it work for you?

This lint was temporarily switched to "allow" because the clippy implementation that was uplifted was linting on expressions that are UB only with Stack Borrow and Tree Borrow and since none of these are official, it was decided to made to temporarily "allow" the lint while fixing it for the next release.

Reply:
> Does it work for you?

I haven't tried it. I'm only trying to describe what we're releasing.

Is it sufficient to just say it was allowed "due to potential false positives"? I think that covers "things that may be UB but aren't decided yet," i.e. SB / TB.

Reply:
Certainly, that's fine by me.

Reply:
Oh, I think I misunderstood that you were suggesting text to add -- but I think brevity is fine here. Thanks!